python/lsst/obs/subaru/gen3/hsc/instrument.py

```diff
@@ -111,17 +128,59 @@ def getCamera(self):
             pupilFactoryClass=HscPupilFactory
         )
 
-    def writeCamera(self, butler):
-        """Write a 'camera' Dataset to the given Butler with an infinite
-        validity range.
+    def getBrighterFatterKernel(self):
+        """Return the brighter-fatter kernel for HSC as a `numpy.ndarray`.
 
         This is a temporary API that should go away once obs_ packages have
-        a standardized approach to writing versioned cameras to a Gen3 repo.
+        a standardized approach to writing versioned kernels to a Gen3 repo.
+        """
+        path = os.path.join(getPackageDir("obs_subaru"), "hsc", "brighter_fatter_kernel.pkl")
+        with open(path, "rb") as fd:
+            kernel = pickle.load(fd, encoding='latin1')  # encoding for pickle written with Python 2
+        return kernel
+
+    def writeCuratedCalibrations(self, butler):
+        """Write human-curated calibration Datasets to the given Butler with
+        the appropriate validity ranges.
+
+        This is a temporary API that should go away once obs_ packages have
+        a standardized approach to this problem.
         """
+
+        # Write cameraGeom.Camera, with an infinite validity range.
         datasetType = DatasetType("camera", ("Instrument", "ExposureRange"), "TablePersistableCamera")
         butler.registry.registerDatasetType(datasetType)
         camera = self.getCamera()
         butler.put(camera, datasetType,
                    instrument=self.getName(),
                    valid_first=datetime.min,
                    valid_last=datetime.max)
+
+        # Write brighter-fatter kernel, with an infinite validity range.
+        datasetType = DatasetType("bfKernel", ("Instrument", "ExposureRange"), "NumpyArray")
+        butler.registry.registerDatasetType(datasetType)
+        # Load and then put instead of just moving the file in part to ensure
+        # the version in-repo is written with Python 3 and does need
+        # `encoding='latin1'` to be read.
+        bfKernel = self.getBrighterFatterKernel()
+        butler.put(bfKernel, datasetType,
+                   instrument=self.getName(),
+                   valid_first=datetime.min,
+                   valid_last=datetime.max)
```

Comment:
In Gen3 the validity ranges are just data ID key/value pairs (though that may change in the future).  Those are stored in the registry database (`DatasetTypeDimensions` table), which I expect to mostly be populated by concrete `PipelineTask`s that define these in their `ConfigClasses` (though I expect it to be configuration that's very rarely changed).  For these particular `DatasetTypes` they're defined right here, with lines like
```
butler.registry.registerDatasetType(datasetType)
```
(it's the `ExposureRange` in the definition of the DatasetType that means that `valid_first` and `valid_last` exist as keys - the fact that there are two names here for the same concept is something @natelust has noted as a problem and something I intend to fix before this gets into broad use).

I do think that's a bit of a step back from Gen2 in terms of having a more-or-less centralized place to go look for DatasetType definitions.  But of course that decentralization is exactly what we need in order to avoid needing to have to modify something like an obs_base paf file every time you introduce a new DatasetType.

Reply:
OK.  Fair enough.